Library/Homebrew/utils/gems.rb

```diff
@@ -51,8 +51,14 @@ def odie_if_defined(message)
   end
 
   def setup_gem_environment!(gem_home: nil, gem_bindir: nil, setup_path: true)
+    # When `HOMEBREW_STACKPROF` is set, this method is called before `HOMEBREW_LIBRARY_PATH` is defined
+    library_path = if ENV["HOMEBREW_STACKPROF"]
```

Comment:
I'd be tempted to just remove this conditional entirely and always use the `HOMEBREW_STACKPROF` version and `raise` if `HOMEBREW_LIBRARY` isn't set.